x-pack/plugins/cases/public/components/add_file/index.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import React, { useCallback, useState } from 'react';
+
+import type { UploadedFile } from '@kbn/shared-ux-file-upload/src/file_upload';
+
+import { FileUpload } from '@kbn/shared-ux-file-upload';
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+
+import { APP_ID, CommentType, ExternalReferenceStorageType } from '../../../common';
+import { CASES_FILE_KINDS } from '../../files';
+import { useKibana } from '../../common/lib/kibana';
+import { useCreateAttachments } from '../../containers/use_create_attachments';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import * as i18n from './translations';
+
+interface AddFileProps {
+  caseId: string;
+  onFileAdded: () => void;
+}
+
+const FILE_ATTACHMENT_TYPE = '.files';
+
+const AddFileComponent: React.FC<AddFileProps> = ({ caseId, onFileAdded }) => {
+  const { notifications } = useKibana().services;
+  const { client: filesClient } = useFilesContext();
+
+  const { owner } = useCasesContext();
+
+  const { isLoading, createAttachments } = useCreateAttachments();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = () => setIsModalVisible(true);
+
+  const onUploadDone = useCallback(
+    async (chosenFiles: UploadedFile[]) => {
+      const file = chosenFiles[0];
+
+      try {
+        await createAttachments({
+          caseId,
+          caseOwner: owner[0],
+          data: [
+            {
+              type: CommentType.externalReference,
+              externalReferenceId: file.id,
+              externalReferenceStorage: {
+                type: ExternalReferenceStorageType.savedObject,
+                soType: FILE_ATTACHMENT_TYPE,
```

Comment:
The `soType` is different from the `externalReferenceAttachmentTypeId`. It should be `FILE_SO_TYPE` from `src/plugins/files/server/saved_objects/file.ts`